Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20157 Support for Eigen thread pool
* #20087 Native ATen/Parallel backend
* #20057 Split ATen/Parallel into interface and backend
* #20050 Move inter-op settings into ATen/Parallel
* #20043 Port ATen/native to ATen/Parallel
* #20032 Port THNN to ATen/Parallel
* #20002 Remove explicit checks for parallelism from TH
* #19105 Port TH library to ATen/Parallel
* #19997 Intra-op parallel microbenchmarks for PT
* **#19993 Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks**
* #19980 Initialize Caffe2 only when running Caffe2 benchmarks
* #19749 Allow a non-OpenMP based build

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

Differential Revision: [D15230363](https://our.internmc.facebook.com/intern/diff/D15230363)

Commit:
Support for non-contiguous tensors in PT benchmarks

Summary:
Add an option to use non-contiguous tensors in PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch --run_mode long

Commit:
Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

Commit:
Update base for Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

dzhulgakov commented on benchmarks/operator_benchmark/benchmark_pytorch.py:
```diff
@@ -17,7 +17,28 @@
+        tensor_shape = list(shape)
+        if not is_contig:
+            tensor_shape = [s * 2 for s in tensor_shape]
+        if dtype in [torch.float32, torch.float64]:
```
dtype.is_floating_point

ilia-cher replied:
I actually want to avoid `float16` because `torch.rand` doesn't work with it, that's why there's explicit check for `torch.float32` and `torch.float64` only

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head

Commit:
Update on "Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks"

Support for non-contiguous tensors and arbitrary dtypes in PT benchmarks

Summary:
Adding support for non-contiguous inputs or arbitrary types in
PT benchmarks

Test Plan:
BUILD_BINARY=1 BLAS=MKL USE_MKLDNN=1 USE_OPENCV=1 USE_FFMPEG=1 python setup.py develop --cmake
cd benchmarks
python -m operator_benchmark.ops.matmul_test --framework PyTorch

gh-metadata: pytorch pytorch 19993 gh/ilia-cher/8/head
